Detections/AzureDevOps/AzDOServiceConnectionUsage.yaml

```diff
@@ -0,0 +1,37 @@
+id: d564ff12-8f53-41b8-8649-44f76b37b99f
+name: Azure DevOps Service Conection Abuse
+description: |
+  'This detection flags builds/releases that use a large number of service connections if they aren't manually allowlisted.
+  This is to determine if someone is hijacking a build/release and adding many service connections in order to abuse or dump credentials from service connections.'
+severity: High
+queryFrequency: 2h
+queryPeriod: 14d
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - Execution
+  - Impact
+relevantTechniques:
+  - T1098
+  - T1527
+  - T1003
+query: |
+
+  let timeframe = 14d;
+  // How many greater than Service Connections you want to view per build/release
+  let ServiceConnectionThreshold = 4;
+  let BypassDefIds = datatable(DefId:string, Type:string, ProjectName:string)
+  [
+  //"103", "Release", "ProjectA",
+  //"42", "Release", "ProjectB",
+  //"122", "Build", "ProjectB"
+  ];
+  AzureDevOpsAuditing
+  | where TimeGenerated > ago(timeframe)
+  | where OperationName == "Library.ServiceConnectionExecuted" 
+  | extend DefId = tostring(Data.DefinitionId), Type = tostring(Data.PlanType), ConnectionId = tostring(Data.ConnectionId)
+  | parse ScopeDisplayName with OrganizationName ' (Organization)'
+  | summarize CurrentCount = dcount(tostring(ConnectionId)), ConnectionNames = make_set(tostring(Data.ConnectionName)) by OrganizationName, tostring(DefId), tostring(Type), ProjectId, ProjectName
```

Comment:
We will need a time to pass into timetamp at the end, it is minimum required for the rule templates. To get this, add in StartTime = min(TimeGenerated), EndTime = max(TimeGenerated) to the summarize before the "by"

Reply:
fixed